components/src/AppHeader/appheader.main.tsx

```diff
@@ -277,15 +324,33 @@ class AppHeaderMain extends React.Component<AppHeaderMainProps, AppHeaderMainSta
             </div>
 
             {(!Config.b2b.enable || (Config.b2b.enable && availability)) && (
-              <div className="cart-link-container">
-                <Link className="cart-link" to={appHeaderLinks.myBag}>
-                  {cartData && cartData['total-quantity'] !== 0 && !isLoading && (
-                    <span className="cart-link-counter">
-                      {cartData['total-quantity']}
-                    </span>
-                  )}
-                  {intl.get('shopping-bag-nav')}
-                </Link>
+              <div>
+                {(!multiCartData || !isLoggedInUser) ? (
+                  <div className="cart-link-container">
+                    <Link className="cart-link" to={appHeaderLinks.myCart}>
+                      {cartData && totalQuantity !== 0 && !isLoading && (
+                      <span className="cart-link-counter">
+                        {totalQuantity}
+                      </span>
+                      )}
+                      {intl.get('shopping-cart-nav')}
+                    </Link>
+                  </div>
+                ) : (
+                  <div className={`cart-link-container multi-cart-dropdown dropdown ${multiCartModalOpened ? 'show' : ''}`}>
+                    <button className={`cart-link dropdown-toggle ${multiCartModalOpened ? 'modal-arrow' : ''}`} id="header_navbar_loggedIn_button" type="button" data-toggle="dropdown">
+                      {cartData && totalQuantity !== 0 && !isLoading && (
+                        <span className="cart-link-counter">
```

Comment:
We discussed in design review. If we're in multi-cart, we should *not* show the counter. It would not make sense as it will show the total # of products, and is no longer cart-specific.